behave/features/steps/means_test.py

```diff
@@ -1,89 +1,144 @@
-    radio_input.click()
-    assert radio_input.get_attribute("checked") == "true"
 
+def set_income_input_field_by_label_for(context, label, value):
+    # At the bottom of case details page under the income inner tab
+    # Can not get element via text as it's broken up.
+    # No idea to target but id string is being used in label for attribute.
+    context.helperfunc.find_by_xpath(f"//label[@for='{label}']/input").send_keys(value)
 
-@when(u'I am on universal credit benefits')
-def step_impl(context):
-    radio_input = context.helperfunc.find_by_css_selector("input[name='your_details-specific_benefits-universal_credit'][value='true']")
-    radio_input.click()
-    assert radio_input.get_attribute("checked") == "true"
 
-
-@when(u'I move onto Finances inner-tab')
-def step_impl(context):
-    page = context.helperfunc
-    finance_inner_tab = page.find_by_xpath("//*[@id='pills-section-list']/li[2]")
-    assert 'Finances' in finance_inner_tab.text
-    actions = ActionChains(page.driver())
-    actions.move_to_element(finance_inner_tab).perform()
-    finance_inner_tab.click()
-    active_inner_tab = context.helperfunc.find_by_class("Pills-pill.is-active") # Confirm finance inner tab is active
-    assert "Finances" in active_inner_tab.text
+def set_expenses_input_field_by_label(context, label, value):
+    context.helperfunc.find_by_xpath(f"//label/span/span[contains(text(), '{label}')]/../../input").send_keys(value)
 
 
-@when(u'I have no savings in the bank')
-def step_impl(context):
-    label = "How much was in your bank account/building society before your last payment went in? "
-    set_input_field_by_label(context, label, '0')
+def set_input_field_by_label(context, label, value):
+    context.helperfunc.find_by_xpath(f"//span[contains(text(),'{label}')]/../input").send_keys(value)
 
 
-@when(u'I have no investments, shares or ISAs')
-def step_impl(context):
-    label = "Do you have any investments, shares or ISAs? "
-    set_input_field_by_label(context, label, '0')
-
-
-@when(u'I have no valuable items worth over £500 each')
-def step_impl(context):
-    label = "Do you have any valuable items worth over £500 each? "
-    set_input_field_by_label(context, label, '0')
-
-
-@when(u'I have no money owed to me')
+@step(u'I am on the Finances tab with the ‘Details’ sub-tab preselected')
 def step_impl(context):
-    label = "Do you have any money owed to you? "
-    set_input_field_by_label(context, label, '0')
+    context.execute_steps(u'''
+        Given I have created a valid discrimination scope
+        Then I am taken to the Finances tab with the ‘Details’ sub-tab preselected
+    ''')
 
 
-@when(u'I select Save assessment')
+@step(u'I move onto {tab_name} inner-tab')
+def step_impl(context, tab_name):
+    # JSON contains list of inner tab names and indexes
+    tab_json = CLA_CASE_DETAILS_INNER_TAB
+    for key in tab_json:
```

Comment:
Do we need this for loop?
My understanding is that the body of the if statement is executed once per method call
It seems you can get the `index_number` with `CLA_CASE_DETAILS_INNER_TAB[tab_name]` or equivalent